src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1099,15 +1099,35 @@ private void DoGraphAutoLayout(object o)
                 models.Add(edge, UndoRedoRecorder.UserAction.Modification);
             }
 
+            // Link a note to the nearest node
+            foreach (NoteModel note in Model.Notes)
+            {
+                try
+                {
+                    NodeModel nd = Model.Nodes.OrderBy(node =>
```

Comment:
Can you not use `graph.Nodes` here instead of `Model.Nodes`? Is it not the same thing. The reason is is so that it will return a `Node` type to you so that you won't have to do a `FindNode` in the following line.